src/pymor/reductors/mt.py

```diff
@@ -0,0 +1,237 @@
+                                    for op in ['A', 'B', 'C', 'D', 'E']})
+        else:
+            fom = self.fom
+
+        self.V = fom.A.source.empty(reserve=r)
+        self.W = fom.A.source.empty(reserve=r)
+
+        if decomposition == 'eig':
+            if fom.order >= sparse_min_size():
+                if not isinstance(fom.A, NumpyMatrixOperator) or fom.A.sparse:
+                    self.logger.warning('Converting operator A to a NumPy array.')
+                if not isinstance(fom.E, IdentityOperator):
+                    if not isinstance(fom.E, NumpyMatrixOperator) or fom.E.sparse:
+                        self.logger.warning('Converting operator E to a NumPy array.')
+            A = to_matrix(fom.A, format='dense')
+            E = None if isinstance(fom.E, IdentityOperator) else to_matrix(fom.E, format='dense')
+
+            if symmetric:
+                poles, ev_r = spla.eig(A, E, right=True)
+                rev = fom.A.source.from_numpy(ev_r.T)
+                lev = rev.copy()
+            else:
+                poles, ev_l, ev_r = spla.eig(A, E, left=True, right=True)
+                rev = fom.A.source.from_numpy(ev_r.T)
+                lev = fom.A.source.from_numpy(ev_l.T)
+            if which == 'SM':
+                idx = np.argsort(np.abs(poles))
+                dominance = np.abs(poles)
+            elif which == 'LR':
+                idx = np.argsort(-poles.real)
+                dominance = np.ones(len(poles))
+            else:
+                absres = np.empty(len(poles))
+                for i in range(len(poles)):
+                    lev[i].scal(1 / lev[i].inner(fom.E.apply(rev[i]))[0][0])
+                    b_norm = fom.B.apply_adjoint(lev[i]).norm()
+                    c_norm = fom.C.apply(rev[i]).norm()
+                    absres[i] = b_norm @ c_norm
+                if which == 'NR':
+                    dominance = -(absres / np.abs(np.real(poles)))
+                    idx = np.argsort(dominance)
+
+                elif which == 'NS':
+                    dominance = -(absres / np.abs(poles))
+                    idx = np.argsort(dominance)
+
+                elif which == 'NM':
+                    idx = np.argsort(-absres)
+                    dominance = -np.array(absres)
+
+                poles = poles[idx]
+                rev = rev[idx]
+                lev = lev[idx]
+                dominance = dominance[idx]
+
+                poles = poles[:2*r]
+                lev = lev[:2*r]
+                rev = rev[:2*r]
+                dominance = dominance[:2*r]
+        elif decomposition == 'samdp':
+            poles, res, rev, lev = samdp(fom.A, fom.E,
+                                         fom.B.as_range_array(),
+                                         fom.C.as_source_array(),
+                                         r, **method_options)
+            absres = [spla.norm(res[i], ord=2) for i in range(len(poles))]
```

Comment:
Using the `axis` argument of the `norm` should do the same.
```suggestion
            absres = spla.norm(res, axis=(1, 2), ord=2)
```